src/components/CreateItem.jsx

```diff
@@ -9,19 +10,37 @@ function CreateItem() {
     frequency: '7',
   };
   const [item, setItem] = useState(itemObj);
+  const token = localStorage.getItem('token');
+
+  const [value, loading, error] = useCollection(
+    db.collection('items').where('token', '==', token),
+  );
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    const token = localStorage.getItem('token');
 
-    db.collection('items').add({
-      name: item.itemName,
-      frequency: item.frequency,
-      lastPurchasedDate: null,
-      date: Date().toLocaleString(),
-      token,
-    });
-    setItem(itemObj);
+    const findDuplicateItem = (item) => {
+      let existingItem = false;
+      value.docs.map((doc) => {
+        if (doc.data().name === item.itemName) {
```

Comment:
Well done with pulling the data out of the doc with the data() method! I found some of these methods difficult to research in the docs, so well done, this works really well 🙌🏻